File: projects/engineering.purdue.edu/kak/programmingwithobjects/pwocode/chap17/FirstWindow.java
// This code example is from the following source:
//
// Book Title:  Programming with Objects, A Comparative Presentation
//              of Object-Oriented Programming with C++ and Java
//
// Book Author:  Avinash Kak
//
// Chapter:     Chapter 17  OO For Graphical User Interfaces, A Tour Of Three Toolkits
//
// Section:     Section 17.5  Minimalist GUI Programs In AWT/Swing
//




//FirstWindow.java

import javax.swing.*;
 
class FirstWindow {
    public static void main(String[] args)  {
        JFrame f = new JFrame( "FirstWindow" );             
        f.setSize( 300, 200 );               
        f.setLocation( 200, 300 );             
        f.setVisible( true );                            
    }
}

File: projects/demos/engineering.purdue.edu/programmingwithobjects/pwocode/chap17/FirstWindow.cc
// This code example is from the following source:
//
// Book Title:  Programming with Objects, A Comparative Presentation
//              of Object-Oriented Programming with C++ and Java
//
// Book Author:  Avinash Kak
//
// Chapter:     Chapter 17  OO For Graphical User Interfaces, A Tour Of Three Toolkits
//
// Section:     Section 17.6  Minimalist GuI Programs In QT
//




//FirstWindow.cc

#include <qapplication.h>
#include <qmainwindow.h>

int main( int argc, char **argv )                                 //(A)
{
    QApplication myApp( argc, argv );                             //(B)

    QMainWindow* myWin = new QMainWindow( 0, 0, 0 );              //(C)
    myWin->resize( 500, 300 );                                    //(D)
    myWin->move( 200, 100 );                                      //(E)

    myApp.setMainWidget(myWin );                                  //(F)
    myWin->show();                                                //(G)
    return myApp.exec();                                          //(H)
}
